### PR TITLE
Fix scheduling of dimension-invariant aliasing expressions

### PR DESCRIPTION
**Pull request by FabioLuporini**

### CODE REVIEW

File: tests/test_dse.py [hunk 1]
```diff
@@ -475,6 +475,49 @@ def test_uncontracted_shape(self):
...
+        """
+        grid = Grid(shape=(6, 6, 6))
+        x, y, z = grid.dimensions  # noqa
+        t = grid.stepping_dim
+
+        f = Function(name='f', grid=grid)
+        f.data_with_halo[:] =\
+            np.linspace(-1, 1, f.data_with_halo.size).reshape(*f.shape_with_halo)
+        u = TimeFunction(name='u', grid=grid, space_order=3)
+        u.data_with_halo[:] = 0.5
```
Comment (mloubout) [inline tests/test_dse.py:490]: any reason to add valeu to halo?

File: tests/test_dse.py [hunk 2]
```diff
@@ -475,6 +475,49 @@ def test_uncontracted_shape(self):
...
+        assert a.symbolic_shape[0] == x_size
+        assert a.symbolic_shape[1] == y_size
+        assert Add(*a.symbolic_shape[2].args) == z_size + 1
+
+        # Check numerical output
+        op0(time_M=1)
+        exp = np.copy(u.data[:])
+        u.data_with_halo[:] = 0.5
+        op1(time_M=1)
+        assert np.allclose(u.data, exp, rtol=10e-7)
```
Comment (mloubout) [inline tests/test_dse.py:518]: I'd add an extra assert on the value of `u` (supposed to e sqrt(2))

File: devito/passes/clusters/aliases.py [hunk 3]
```diff
@@ -699,12 +699,17 @@ def iter(self, ispace):
...
+            # Becomes True as soon as a Dimension in `ispace` is found to
+            # be independent of `intervals`
+            flag = False
             writeto = []
-            for n, i in enumerate(ispace.intervals):
-                interval = mapper.get(i.dim)
-                if interval is not None:
-                    if not writeto and interval == interval.zero():
+            for i in ispace.intervals:
+                try:
```
Comment (mloubout) [inline devito/passes/clusters/aliases.py:710]:
```
interval = mapper.get(i.dim)
if not interval:
    if any(i.dim in d._defines for d in mapper):
         # E.g., `i.dim` is `x0_blk0` in `x0_blk0[0,0]<0>`
          pass
     else:
           # E.g., `t[0,0]<0>` in the case of t-invariant aliases
          flag = True
```

would avoid large try/except.

File: tests/test_dse.py (hunk 1 above)
Comment (FabioLuporini) [reply tests/test_dse.py:490]: not in this example, just copy-paste from the other tests. I can tweak it

File: tests/test_dse.py (hunk 2 above)
Comment (FabioLuporini) [reply tests/test_dse.py:518]: yep, I can add an extra assert -- but what "value of u" are you specifically referring to?

File: devito/passes/clusters/aliases.py (hunk 3 above)
Comment (FabioLuporini) [reply devito/passes/clusters/aliases.py:710]: Hang on, I'll try to make it nicer

Comment (FabioLuporini) [reply devito/passes/clusters/aliases.py:710]: now?

Comment (mloubout) [reply devito/passes/clusters/aliases.py:710]: but do we really need the try/except?

Comment (FabioLuporini) [reply devito/passes/clusters/aliases.py:710]: it's completely pythonic and I'm sure we use the same pattern somewhere else in Devito

so: "really need" -> no , "is it fine to have" -> yes

Comment (mloubout) [reply devito/passes/clusters/aliases.py:710]: ok gtg

File: tests/test_dse.py (hunk 2 above)
Comment (mloubout) [reply tests/test_dse.py:518]: Sorry, for some reason read `f.data=1` forget what I said.

**Pull request merged**